lib/alp_workloads/kvm_workload_utils.pm

```diff
@@ -0,0 +1,209 @@
+}
+
+sub config_host_and_kvm_container {
+    # Copy contents from kvm container to host to set it up
+    assert_script_run("clear; $ct_engine container runlabel install $kvm_image");
+    save_screenshot;
+
+    # Replace /etc/kvm-container.conf IMAGE value with $kvm_image if non-default
+    if ($kvm_image ne $DEFAULT_KVM_IMAGE) {
+        assert_script_run("cp /etc/kvm-container.conf /etc/kvm-container.conf.orig");
+        assert_script_run(q@sed -i 's#^\s*IMAGE=.*$#IMAGE=@ . $kvm_image . q@#g' /etc/kvm-container.conf@);
+        assert_script_run("grep ^IMAGE= /etc/kvm-container.conf");
+        save_screenshot;
+        record_info("Change /etc/kvm-container.conf default IMAGE!", "New value is IMAGE=$kvm_image.");
+    }
+
+    # Share /dev/sshserial from host to kvm container for let openqa root-ssh console works
+    assert_script_run("cp /usr/local/bin/kvm-container-manage.sh /usr/local/bin/kvm-container-manage.sh.orig");
+    assert_script_run(q@sed -i 's#podman create#podman create --volume /dev/sshserial:/dev/sshserial #g' /usr/local/bin/kvm-container-manage.sh@);
+    script_run("diff /usr/local/bin/kvm-container-manage.sh /usr/local/bin/kvm-container-manage.sh.orig");
+    save_screenshot;
+    record_info("kvm-container-manage.sh has been changed to share sol console needed device with kvm container.");
+    assert_script_run("kvm-container-manage.sh create");
+    save_screenshot;
+    record_info("Create KVM container successfully!", script_output("$ct_engine ps --all"));
+}
+
+sub cleanup_host_and_kvm_container {
+    assert_script_run("if $ct_engine ps | grep libvirtd; then kvm-container-manage.sh stop;fi");
+    assert_script_run("if $ct_engine ps --all | grep libvirtd; then kvm-container-manage.sh rm;fi");
+    assert_script_run("if which kvm-container-manage.sh;then kvm-container-manage.sh uninstall;fi");
+    validate_script_output("$ct_engine ps --all", sub { $_ !~ 'libvirtd' });
+    save_screenshot;
+    record_info("Remove KVM container and related files from host successfully!");
+}
+
+sub start_kvm_container {
+    assert_script_run("$ct_engine start libvirtd");
+    save_screenshot;
+    assert_script_run("$ct_engine ps | grep -i libvirtd | grep Up");
+    save_screenshot;
+    record_info("Start kvm container successfully!");
+}
+
+sub enter_kvm_container_sh {
+    enter_cmd("clear;$ct_engine exec -ti libvirtd bash", 15);    # interactive mode, so no return immediately
+    save_screenshot;
+    assert_screen('in-libvirtd-container-bash');
+    assert_script_run("echo Hello from libvirtd container");
+    save_screenshot;
+
+    assert_script_run("ip a");
+    save_screenshot;
+    assert_script_run("ip route show all");
+    save_screenshot;
+    assert_script_run("virsh list --all");
+    record_info('Please check if it is from within libvirtd container');
+}
+
+sub exit_kvm_container {
+    assert_script_run("clear");
```

Comment:
Quick question, why use with `assert_script_run("clear"); ` here?

Reply:
Assert_screen is used to ensure/check whether the container exit/entrance is successful or not. Clear can help clean up screens first, which ensures that the latter screenshot check the needed areas, rather than earlier screen output.